name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/SecurityCache.java

```diff
@@ -32,18 +32,20 @@ public SecurityCache(Client client)
     {
         this.client = client;
 
-        this.localMaps.add(client.getSecurities().stream().filter(s -> s.getIsin() != null && !s.getIsin().isEmpty())
+        this.localMaps.add(client.getSecurities().stream()
+                        .filter(s -> s.getIsin() != null && !s.getIsin().isEmpty() && !s.isRetired())
                         .collect(Collectors.toMap(Security::getIsin, s -> s, (l, r) -> DUPLICATE_SECURITY_MARKER)));
```

Comment:
Instead of the existing merge function ```(l, r) -> DUPLICATE_SECURITY_MARKER ```, we could check for the retired flag there.

```

(l, r) -> {
   if (l.isRetired() ^ r.isRetired()) {
     return !l.isRetired() ? l : r;
  }
  else return DUPLICATE_SECURITY_MARKER;
}

```

Maybe we would also have to check whether l or r is a DUPLICATE_SECURITY_MARKER. 

(I am not at my laptop so it is just jotted down to share my thinking)